Add-ons/UmbracoForms/Developer/Working-With-Data/index.md

```diff
@@ -1,106 +1,113 @@
-DynamicRecordList GetRecordsFromPage(int pageId)
+PagedResult<IRecord> GetRecordsFromPage(int pageId, int pageNumber, int pageSize)
 ```
 
 Returns all records from all forms on the Umbraco page with the id = `pageId` as a DynamicRecordList.
 
 ### GetRecordsFromFormOnPage
 
 ```csharp
-DynamicRecordList GetRecordsFromFormOnPage(int pageId, string formId)
+PagedResult<IRecord> GetRecordsFromFormOnPage(int pageId, string formId, int pageNumber, int pageSize)
 ```
 
 Returns all records from the form with the id = `formId` on the Umbraco page with the id = `pageId` as a DynamicRecordList.
 
 ### GetRecordsFromForm
 
 ```csharp
-DynamicRecordList GetRecordsFromForm(string formId)
+PagedResult<IRecord> GetRecordsFromForm(string formId, int pageNumber, int pageSize)
 ```
 
 Returns all records from the form with the ID = formId as a DynamicRecordList
 
 ## DynamicRecordsList and DynamicRecord
 
-All of these methods will return an object of type `DynamicRecordList` so you can iterate through the DynamicRecord objects.
+All of these methods will return an object of type `PagedResult<IRecord>` so you can iterate through the `IRecord` objects.
 
-The properties available on a DynamicRecord are:
+The properties available on a `IRecord` are:
 
 ```csharp
+int Id
+FormState State
 DateTime Created
-string Form
-string Id
+DateTime Updated
+Guid Form
 string IP
-object MemberKey
-Dictionary<Guid, RecordField> RecordFields
-FormState? State
 int UmbracoPageId
-DateTime Updated
+string MemberKey
+Guid UniqueId
+Dictionary<Guid, RecordField> RecordFields
 ```
 
-In order to access custom form fields you can use the dot notation, using the field caption but removing all spaces and non alphanumeric characters.
+In order to access custom form fields, these are available in the `RecordFields` property. 
+Furthermore there exists an extension method named `ValueAsString` on  `IRecord` in `Umbraco.Forms.Core.Services`, such that you can get the value as string given the alias of the field.
+
+This extension method handle multi value fields by comma separating the values. E.g. "A, B, C"
 
 ## Sample razor script 
 
 Sample script that is outputting comments using a form created with the default comment form template.
 
 ```csharp
-@using Umbraco.Forms.Mvc.DynamicObjects
-
+@using Umbraco.Core;
+@using Umbraco.Core.Composing
+@using Umbraco.Forms.Core.Services
+@{
+   var recordReaderService = Current.Factory.GetInstance<IRecordReaderService>();
+}
 <ul id="comments">
-	@foreach (dynamic record in Library
-			.GetApprovedRecordsFromPage(@CurrentPage.Id))
-	{
-		<li>
-			@record.Created.ToString("dd MMMM yyy")
-			@if(string.IsNullOrEmpty(record.Website)){
-				<strong>@record.Name</strong>
-			}
-			else{
-				<strong>
-				<a href="@record.Website" target="_blank">@record.Name</a>
-				</strong>
-			}
-			<span>said</span>
-			<p>@record.Comment</p>
-		</li>
-	}
+   @foreach (dynamic record in recordReaderService.GetApprovedRecordsFromPage(Model.Id, 0, 10))
```

Comment:
I don't think we should be using `dynamic` here... or anywhere?  This method also returns `PagedResult<IRecord>` now so the enumeration and usage of this method will need to be different

Reply:
Sure.. Thats a bug.. `dynamic` should be `var`